pkg/controller/user-cluster-controller-manager/resources/resources/mla/promtail/daemonset.go

```diff
@@ -229,6 +230,22 @@ func DaemonSetCreator() reconciling.NamedDaemonSetCreatorGetter {
 					},
 				},
 			}
+
+			defResourceRequirements := map[string]*corev1.ResourceRequirements{
+				containerName: defaultResourceRequirements.DeepCopy(),
+			}
+			var err error
+			if overrides == nil {
+				err = resources.SetResourceRequirements(ds.Spec.Template.Spec.Containers, defResourceRequirements, nil, ds.Annotations)
+			} else {
+				overridesRequirements := map[string]*corev1.ResourceRequirements{
+					containerName: overrides.DeepCopy(),
+				}
+				err = resources.SetResourceRequirements(ds.Spec.Template.Spec.Containers, defResourceRequirements, overridesRequirements, ds.Annotations)
+			}
```

Comment:
Why do you have this check instead of just calling `SetResourceRequirements`? You could do something like this, which is a little bit simpler and easier to read:

```suggestion
			var overridesRequirements map[string]*corev1.ResourceRequirements
			if overrides != nil {
				overridesRequirements = map[string]*corev1.ResourceRequirements{
					containerName: overrides.DeepCopy(),
				}
			}
			err = resources.SetResourceRequirements(ds.Spec.Template.Spec.Containers, defResourceRequirements, overridesRequirements, ds.Annotations)
```